e2e/tests/taxonomy.test.js

```diff
@@ -722,3 +722,240 @@ Scenario('Taxonomy maxUsages', async ({ I, LabelStudio, AtTaxonomy }) => {
+  AtTaxonomy.toggleGroupWithText('Four to seven');
+  AtTaxonomy.seeItemByText('Two');
+  AtTaxonomy.seeItemByText('Five');
+  AtTaxonomy.clickItemByText('Three');
+  AtTaxonomy.clickItemByText('Four');
+  AtTaxonomy.seeSelectedValues(['Three', 'Four']);
+  AtTaxonomy.clickTaxonomy();
+  I.click('Eight'); // click on the choice
+  I.seeElement('.ant-checkbox-checked [name=\'Eight\']');
+  I.say('Should get results for taxonomy and choices');
+
+  let result = await LabelStudio.serialize();
+
+  assert.deepStrictEqual(result[0].value.taxonomy, [['One to three', 'Three'], ['Four to seven', 'Four']]);
+  assert.deepStrictEqual(result[1].value.choices, ['Eight']);
+
+  I.say('Should get results for only taxonomy when visibleWhen is not met');
+  AtTaxonomy.clickTaxonomy();
+  AtTaxonomy.clickItemByText('Four');
+  AtTaxonomy.clickTaxonomy();
+  I.dontSeeElement('.ant-checkbox-checked [name=\'Eight\']');
+
+  result = await LabelStudio.serialize();
+
+  assert.deepStrictEqual(result[0].value.taxonomy, [['One to three', 'Three']]);
+  assert.deepStrictEqual(result?.[1]?.value?.choices, undefined);
+
+  I.say('Should get results for taxonomy and choices when visibleWhen is met');
+  AtTaxonomy.clickTaxonomy();
+  AtTaxonomy.clickItemByText('Four');
+  AtTaxonomy.clickTaxonomy();
+  I.seeElement('.ant-checkbox-checked [name=\'Eight\']');
+
+  result = await LabelStudio.serialize();
+
+  assert.deepStrictEqual(result[0].value.taxonomy, [['One to three', 'Three'], ['Four to seven', 'Four']]);
+  assert.deepStrictEqual(result[1].value.choices, ['Eight']);
+
+  await session('Deserialization', async () => {
+    I.amOnPage('/');
+    LabelStudio.init({
+      config: createConfig(),
+      data: {
+        text: 'A text',
+      },
+      annotations: [{
+        id: 'test',
+        result,
+      }],
+    });
+    I.say('Should see the same result');
+    AtTaxonomy.clickTaxonomy();
+    AtTaxonomy.toggleGroupWithText('One to three');
+    AtTaxonomy.toggleGroupWithText('Four to seven');
+    AtTaxonomy.seeCheckedItemByText('Three');
+    AtTaxonomy.seeCheckedItemByText('Four');
+    AtTaxonomy.seeSelectedValues(['Three', 'Four']);
+    I.seeElement('.ant-checkbox-checked [name=\'Eight\']');
+  });
+
+  await session('ShowFullPath', async () => {
+    //showFullPath
+    I.amOnPage('/');
+    LabelStudio.init({
+      config: createConfig({ showFullPath: true }),
+      data: {
+        text: 'A text',
+      },
+      annotations: [{
+        id: 'test',
+        result,
+      }],
+    });
+    I.say('Should see the full paths');
+    AtTaxonomy.clickTaxonomy();
+    AtTaxonomy.seeSelectedValues(['One to three / Three', 'Four to seven / Four']);
+    I.seeElement('.ant-checkbox-checked [name=\'Eight\']');
+  });
+});
+
+Scenario('Taxonomy visibleWhen with aliases', async ({ I, LabelStudio, AtTaxonomy }) => {
+  const createConfig = ({ showFullPath = false, visibleWhen = 'choice-selected', whenChoiceValue = 'Four' } = {}) => `

```

Comment:
I don't think this is valid config — aliases should replace value completely if given afaik, so referring to this exact Choice should be only by alias — `4`. Do we have other requirements?

Reply:
It replaces the value completely, but the label is still based on the value prop. So selections and targeting will still be based on the label `Four` but values in results and elsewhere is `4`. The assertions are driven by the label text from what I can see. Is this not correct? I based this on the other tests with regards to aliases.